src/Symfony/Component/Notifier/Message/WebPushMessage.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Message;
+
+use Symfony\Component\Notifier\Notification\Notification;
+
+/**
+ * @author Tomas Norkūnas <norkunas.tom@gmail.com>
+ */
+final class WebPushMessage implements MessageInterface
+{
+    private $transport;
+    private $subject;
+    private $content;
+    private $options;
+    private $notification;
+
+    public function __construct(string $subject, string $content, MessageOptionsInterface $options = null)
+    {
+        $this->subject = $subject;
+        $this->content = $content;
+        $this->options = $options;
+    }
+
+    public static function fromNotification(Notification $notification): self
+    {
+        $message = new self($notification->getSubject(), $notification->getContent());
+        $message->notification = $notification;
+
+        return $message;
+    }
+
+    /**
+     * @return $this
+     */
+    public function subject(string $subject): self
+    {
+        $this->subject = $subject;
+
+        return $this;
+    }
+
+    public function getSubject(): string
+    {
+        return $this->subject;
+    }
+
+    public function content(string $content): self
+    {
+        $this->content = $content;
+
+        return $this;
+    }
+
+    public function getContent(): string
+    {
+        return $this->content;
+    }
+
+    public function getRecipientId(): ?string
+    {
+        return $this->options ? $this->options->getRecipientId() : null;
+    }
+
+    /**
+     * @return $this
+     */
+    public function options(MessageOptionsInterface $options): self
+    {
+        $this->options = $options;
+
+        return $this;
+    }
+
+    public function getOptions(): ?MessageOptionsInterface
+    {
+        return $this->options;
+    }
+
+    /**
+     * @return $this
+     */
+    public function transport(?string $transport): self
+    {
+        $this->transport = $transport;
+
+        return $this;
+    }
+
+    public function getTransport(): ?string
+    {
+        return $this->transport;
+    }
+
+    public function getNotification(): ?Notification
```

Comment:
Why is this needed?

Reply:
Probably copied from ChatMessage, so don't know really if it's needed :)

Reply:
Lets keep it like this 👍